stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/StreamFileUtil.kt

```diff
@@ -54,4 +61,144 @@ public object StreamFileUtil {
+     * file operations.
+     *
+     * @return Returns a [Result]. If the action was successful
+     * [Result.data] will contain a [File] pointing to the cache directory,
+     * otherwise [Result.error] will contain a [ChatError].
+     */
+    @Suppress("TooGenericExceptionCaught")
+    private fun getOrCreateStreamCacheDir(
+        context: Context,
+    ): Result<File> {
+        return try {
+            val file = File(context.cacheDir, STREAM_CACHE_DIR_NAME).also { streamCacheDir ->
+                streamCacheDir.mkdirs()
+            }
+
+            Result(data = file)
+        } catch (e: Exception) {
+            val chatError = ChatError(
+                message = "Could not get or create the Stream cache directory",
+                cause = e
+            )
+
+            Result(error = chatError)
+        }
+    }
+
+    /**
+     * Deletes all the content contained within the
+     * Stream cache directory.
+     *
+     * @param context The [Context] necessary to perform
+     * file operations.
+     *
+     * @return Returns a [Result]. If the action was successful
+     * [Result.data] will contain [Unit], otherwise [Result.error]
+     * will contain a [ChatError].
+     */
+    @Suppress("TooGenericExceptionCaught")
+    public fun clearStreamCache(
+        context: Context,
+    ): Result<Unit> {
+        return try {
+            val directory = File(context.cacheDir, STREAM_CACHE_DIR_NAME)
+            directory.deleteRecursively()
+
+            Result(data = Unit)
+        } catch (e: Exception) {
+            val chatError = ChatError(
+                message = "Could clear the Stream cache directory",
+                cause = e
+            )
+
+            Result(error = chatError)
+        }
+    }
+
+    /**
+     * Hashes the links of given attachments and then tries to create a new file
+     * under that hash. If the file already exists checks that the full file
+     * has been written and shares it if it has, in other cases downloads the file
+     * and writes it.
+     *
+     * @param context The Android [Context] used for path resolving and [Uri] fetching.
+     * @param attachment the attachment to be downloaded.
+     *
+     * @return Returns a [Result]. If the action was successful
+     * [Result.data] will contain a [Uri] pointing to the file, otherwise [Result.error]
+     * will contain a [ChatError].
+     */
+    @Suppress("ReturnCount")
+    public suspend fun writeFileToShareableFile(
+        context: Context,
+        attachment: Attachment,
+    ): Result<Uri> {
+        val runCatching = kotlin.runCatching {
+            val getOrCreateCacheDirResult = getOrCreateStreamCacheDir(context)
+            if (getOrCreateCacheDirResult.isError) return Result(error = getOrCreateCacheDirResult.error())
+
+            val streamCacheDir = getOrCreateCacheDirResult.data()
+
+            val attachmentName = (attachment.url ?: attachment.assetUrl)?.hashCode()
```

Comment:
we should name this: nameHashCode or attachmentHash or something.

It doens't make sense to put this in the same line:
```
 val fileName = CACHED_FILE_PREFIX + attachmentName.toString() + attachment.name
```